src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,482 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+using namespace std;
+
+void compute_strides(Rpp32u *strides, Rpp32u *shape, Rpp32u nDim)
+{
+    if (nDim > 0)
+    {
+        uint64_t v = 1;
+        for (int i = nDim - 1; i > 0; i--)
+        {
+            strides[i] = v;
+            v *= shape[i];
+        }
+        strides[0] = v;
+    }
+}
+
+void increment_ndim_ptr(Rpp32f **dstPtr, Rpp32u nDim, Rpp32u increment)
+{
+    for(int i = 0; i < nDim; i++)
+        dstPtr[i] += increment;
+}
+
+void rpp_store16_f32_f32_channelwise(Rpp32f **dstPtr, __m128 *p)
+{
+    _mm_storeu_ps(dstPtr[0], p[0]);
+    _mm_storeu_ps(dstPtr[1], p[1]);
+    _mm_storeu_ps(dstPtr[2], p[2]);
+    _mm_storeu_ps(dstPtr[3], p[3]);
+}
+
+void rpp_transpose4x8_avx(__m256 *pSrc, __m128 *pDst)
```

Comment:
Ideally lets keep same convention of compute_..._avx() inside the rpp_cpu_common.hpp for this